### PR TITLE
refactor: move from pool-per-module to singleton pool with PgAdapter-per-module

### PR DESCRIPTION
**Pull request by NathanFlurry**

### CODE REVIEW

**Comment by NathanFlurry** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/rivet-gg/opengb/411?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#413** <a href="https://app.graphite.dev/github/pr/rivet-gg/opengb/413?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#412** <a href="https://app.graphite.dev/github/pr/rivet-gg/opengb/412?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#411** <a href="https://app.graphite.dev/github/pr/rivet-gg/opengb/411?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#407** <a href="https://app.graphite.dev/github/pr/rivet-gg/opengb/407?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#406** <a href="https://app.graphite.dev/github/pr/rivet-gg/opengb/406?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#405** <a href="https://app.graphite.dev/github/pr/rivet-gg/opengb/405?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @NathanFlurry and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/build/entrypoint.ts [hunk 1]
```diff
@@ -40,7 +40,18 @@ export async function generateEntrypoint(project: Project, opts: BuildOpts) {
...
+		imports += `
+		// Import Prisma adapter for Postgres
+    //
+    // We can't use esm.sh for these because they rely on special Node
+    // functionality & don't need to be portable
+    //
+    // https://github.com/esm-dev/esm.sh/issues/684
+    // import 'npm:pg-cloudflare';
+		import pg from "npm:pg@8.12.0";
+		import { PrismaPg } from "npm:@prisma/adapter-pg@5.12.0";
```
Comment (MasterPtato) [inline src/build/entrypoint.ts:54]: iirc we had a ticket for removing node compatibility from deployed workers, is that canceled now that we need to use node for hyperdrive? or is it handled by esbuild somehow?

Comment (MasterPtato) [inline src/build/entrypoint.ts:54]: iirc we had a ticket for removing node compatibility from deployed workers, is that canceled now that we need to use node for hyperdrive? or is it handled by esbuild somehow?

**Review by MasterPtato — APPROVED** [review]

File: src/build/entrypoint.ts (hunk 1 above)
Comment (NathanFlurry) [reply src/build/entrypoint.ts:54]: good catch. this comment is out of date. i think we're actually polyfilling node for everything now. it actually falls back to a lib called `pg-cloudflare` under the hood.

**Comment by NathanFlurry** [comment]
### Merge activity

* **Jun 25, 12:16 AM EDT**: @NathanFlurry merged this pull request with [Graphite](https://app.graphite.dev/github/pr/rivet-gg/opengb/411).

**Pull request merged**